src/main/java/org/zalando/nakadi/service/timeline/TimelineService.java

```diff
@@ -0,0 +1,156 @@
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, activeTimeline.getTopic(), new Date());
+                switchTimeline(eventType, nextTimeline, () -> nextTimeline.setSwitchedAt(new Date()));
+            } else {
+                final String newTopic = nextTopicRepo.createTopic(partitionStatistics.size(),
+                        eventType.getOptions().getRetentionTime());
+                final Timeline nextTimeline = Timeline.createTimeline(activeTimeline.getEventType(),
+                        activeTimeline.getOrder() + 1, storage, newTopic, new Date());
+                LOG.info("Switching timelines. Topics: {} -> {}", activeTimeline.getTopic(), nextTimeline.getTopic());
+                switchTimeline(eventType, nextTimeline, () -> {
+                    final Timeline.StoragePosition storagePosition =
+                            StoragePositionFactory.createStoragePosition(activeTimeline, currentTopicRepo);
+                    activeTimeline.setLatestPosition(storagePosition);
+                    nextTimeline.setSwitchedAt(new Date());
+                    timelineDbRepository.updateTimelime(activeTimeline);
+                });
+            }
+        } catch (final NakadiException ne) {
+            throw new TimelineException(ne.getMessage(), ne);
+        }
+    }
+
+    public Timeline getTimeline(final EventType eventType) throws TimelineException {
+        try {
+            final String eventTypeName = eventType.getName();
+            final Optional<Timeline> activeTimeline = eventTypeCache.getActiveTimeline(eventTypeName);
+            if (activeTimeline.isPresent()) {
+                return activeTimeline.get();
+            }
+
+            final Storage storage = storageDbRepository.getStorage(DEFAULT_STORAGE)
+                    .orElseThrow(() -> new UnableProcessException("Fake timeline creation failed for event type " +
+                            eventType.getName() + ".No default storage defined"));
+            return Timeline.createFakeTimeline(eventType, storage);
+        } catch (final NakadiException e) {
+            LOG.error("Failed to get timeline for event type {}", eventType.getName(), e);
+            throw new TimelineException("Failed to get timeline", e);
+        }
+    }
+
+    public TopicRepository getTopicRepository(final EventType eventType)
+            throws TopicRepositoryException, TimelineException {
+        final Timeline timeline = getTimeline(eventType);
+        return topicRepositoryHolder.getTopicRepository(timeline.getStorage());
+    }
+
+    private void switchTimeline(final EventType eventType,
+                                final Timeline nextTimeline,
+                                final Runnable switcher) {
+        transactionTemplate.execute(status -> {
```

Comment:
You can check out the details here: https://github.com/spring-projects/spring-framework/blob/21ae169653b9da60f9ec8bfe18f5d1bf9e801c94/spring-tx/src/main/java/org/springframework/transaction/support/TransactionTemplate.java#L125